doc: ug_thingy53: Improve user guide (#5596)
nrfconnect/sdk-nrf: samples/bluetooth/mesh/light/README.rst

   :header: heading
   :rows: nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns, nrf52840dk_nrf52840, nrf52dk_nrf52832, nrf52833dk_nrf52833, nrf52833dk_nrf52820, thingy53_nrf5340_cpuapp

The sample also requires a smartphone with Nordic Semiconductor's nRF Mesh mobile app installed in one of the following versions:

  * `nRF Mesh mobile app for Android`_
  * `nRF Mesh mobile app for iOS`_

.. note::
   |thingy53_sample_note|

> 

Reviewer: This aplay for both the light and light_switch sample: When doing dfu upgrade by the mobile app, from light to light_switch sample or vice a versa. it is important that the device is not provisioned. If the sample is provisioned the sample will not be possible to provision after dfu upgrade and it is not possible to reset the old provisioned device. So it is important to reset the provisioned device (that way unprovision it) before doing a dfu upgrade by the mobil app.
I wrote this here, as I do not know the best way to put this in the doc. This aplay for both the light and ligth_switch sample.

Author: Hi @Balaklaka 
Thanks a lot for your input. Isn't the provisioning data handled in Zephyr's settings storage? In that case it should be by default automatically cleared when DFU is performed. In general we cannot assume that settings will be compatible between various samples or applications, so settings partition should be erased before booting a new application.

Reviewer: I do not necessary think that it is correct to automatically delete the settings when DFU is performed, because if we update to a new version of the same example we want to keep settings.

Author: I think deleting settings by default would be the safest option (if we have no additional information about the firmware that is being swapped). Otherwise we may encounter strange issues.
If I recall correctly, settings cleanup is done using separate mcumgr command, so we may consider introducing additional logic that would prevent deleting settings if this is not necessary (e.g. if we are sure that settings remain compatible).

Reviewer: We had a discussion on this, and it is a to complex fix now. And most likely it has to be fixed upstream. So my suggestion for this release is to add a comment about this for now.

Author: Added short description related to settings erase

Reviewer 2: I will discuss with Wayland and MAG about making the settings erase optional. Enabled by default, but a user can retain the settings partition by un-checking a box or similar.

   :header: heading
   :rows: nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns, nrf52840dk_nrf52840, nrf52dk_nrf52832, nrf52833dk_nrf52833, nrf52833dk_nrf52820, thingy53_nrf5340_cpuapp

The sample also requires a smartphone with Nordic Semiconductor's nRF Mesh mobile app installed in one of the following versions:

  * `nRF Mesh mobile app for Android`_
  * `nRF Mesh mobile app for iOS`_

.. note::

>    If the sample is used with Thingy:53, it complies with Thingy:53 application guide and enables additional features.
>    See :ref:`thingy53_app_guide` for details.

Reviewer 3: These lines should be an entry in shortcuts.txt, to avoid duplicating them for each sample.
Also, I'm not sure how to understand "it complies with Thingy:53 application guide". How can a sample (firmware) comply with an application guide (documentation)?
```suggestion
  If you use this sample with Thingy:53, it enables additional features proper of the Thingy:53 application.
  See :ref:`thingy53_app_guide` for details.
```

Reviewer 3: OK, now I get the meaning. So either use my propose text, which I think fits the case here, or something like `If you use ..., check the <Thingy:53 application guide> for details about the additional features that are enabled by default`. I just don't like the phrase "it complies with the guide" -- the guide is just a help, not a requirement. @wiba-nordic , what do you think?

Reviewer 4: I agree that the phrasing you have in this comment "check the guide for details about the additional features that are enabled by default" is a clear improvement.

Author: I rephrased this one a bit. Would this be ok?

Reviewer 2: Wouldnt it be more correct to state that if you build this sample for the Thingy:53 it enables additional features?

Author: Ok, I will update it

>    See :ref:`_thingy53_app_guide` for details.

Reviewer 4: ```suggestion
   See :ref:`thingy53_app_guide` for details.
```
